fairroot/generators/FairIonGenerator.cxx

```diff
@@ -107,9 +108,7 @@ FairIonGenerator::FairIonGenerator(Int_t z,
   fVy   = vy;
   fVz   = vz;
   */
-    char buffer[20];
-    sprintf(buffer, "FairIon%d", fgNIon);
-    fIon = new FairIon(buffer, z, a, q);
+    fIon = new FairIon(fmt::format("FairIon{}", fgNIon).c_str(), z, a, q);
```

Comment:
Please assign the return value of `fmt::format` to a local variable to make sure that its life time is long enough.

```suggestion
    std::string ionName = fmt::format("FairIon{}", fgNIon)
    fIon = new FairIon(ionName.c_str(), z, a, q);
```

(There are more places. I just used this one.)

Reply:
https://stackoverflow.com/a/10006989 Googled, this claims this should work out.

Reply:
Hmmm.

Then the code should be consistent, please. Either try to avoid local variables wherever possible or use them everywhere, please.